src/milia/api/io.clj

```diff
@@ -123,19 +124,26 @@
   (and (= 401 status) (:temp-token @*credentials*)))
 
 (defn http-request
-  "Send an HTTP request and catch some exceptions."
+  "Send HTTP request and handle exceptions"
   [method url http-options]
-  (let [req-fn #(call-client-method method url (build-req http-options))]
-    (try+  ; Catch all bad statuses
+  (let [send-request
+        #(call-client-method method url (build-req http-options))]
+    (try+
      (client/with-connection-pool
        {:default-per-route (env :jetty-min-threads)
-        :threads (env :jetty-min-threads)
-        :timeout 10}
-       (try+ ; Catch 401 with token expire messages
-        (req-fn)
+        :threads (env :jetty-min-threads)}
+       (try+
+        (send-request)
         (catch #(expired-token? %) response
-          (do
-            (refresh-temp-token)
-            (req-fn)))))
-     ;; To avoid NPE, default to a fake 555 status if no status is returned
-     (catch #(<= 400 (:status % 555)) response response))))
+          (refresh-temp-token)
+          (send-request))
+        (catch NoHttpResponseException _
+          ;; Because Core doesn't respond with a 401 on unauthorized PATCH requests
+          (refresh-temp-token)
```

Comment:
Why did you remove the `do` here? We want to refresh-temp-token then send-request.